# Hypothesis Testing

In [1]:
import numpy as np
import pandas as pd
import scipy.stats as stats

# Test 1

It is assumed that the mean systolic blood pressure is μ = 120 mm Hg. In the Honolulu Heart Study, a sample of n = 100 people had an average systolic blood pressure of 130.1 mm Hg with a standard deviation of 21.21 mm Hg. Is the group significantly different (with respect to systolic blood pressure!) from the regular population?

<b> Set up test

Ho = Null Hypothesis = mean is equal to 120mm Hg (Ho: mu = 120) <br>
Ha = Alternative Hypothesis = mean is not equal to 120mm Hg (Ha: mu != 120)

In [2]:
n = 100
sd = 21.21
mu = 120
sample_mean = 130.1

<b> Compute standard error of mean

In [3]:
sem = sd/np.sqrt(n)
sem

2.121

<b> Find t score

In [4]:
z = (sample_mean - mu)/sem
z

4.761904761904759

<b> Find critical value using a 95% confidence interval (two-tailed test)

In [5]:
zc = stats.norm.ppf(1-(0.05/2))
zc

1.959963984540054

<b> Compare z score with critical value

Since z=4.76 > zc=1.96 we reject the null and assume that the mean is not equal to 120mm Hg

## Test 2

In a packing plant, a machine packs cartons with jars. It is supposed that a new machine will pack faster on the average than the machine currently used. To test that hypothesis, the times it takes each machine to pack ten cartons is recorded. The results, in seconds, are shown in the table. Assume that there is sufficient evidence to conduct the t test, does the data provide sufficient evidence to show if one machine is better than the other?

In [27]:
df = pd.read_csv("Data/Data_Machine.txt")
df[['New Machine','Old Machine']] = df['New Machine\t  Old Machine'].str.split('\t', expand=True)
df = df.drop(columns='New Machine\t  Old Machine')

In [28]:
df

New Machine     Old Machine
0        42.1            42.7
1          41            43.6
2        41.3            43.8
3        41.8            43.3
4        42.4            42.5
5        42.8            43.5
6        43.2            43.1
7        42.3            41.7
8        41.8              44
9        42.7            44.1